jetstream/statistics.py

```diff
@@ -422,6 +423,56 @@ def transform(
         )
 
 
+@attr.s(auto_attribs=True)
+class LinearModelMean(Statistic):
+    drop_highest: float = attr.field(default=0.005, validator=attr.validators.instance_of(float))
+    covariate_adjustment: dict[str, dict[str, str]] | None = attr.field(default=None)
+
+    @covariate_adjustment.validator
+    def check(self, attribute, value):
+        if value is not None:
+            covariate_period = parser_metric.AnalysisPeriod(value["period"])
+            if covariate_period not in (
+                parser_metric.AnalysisPeriod.PREENROLLMENT_WEEK,
+                parser_metric.AnalysisPeriod.PREENROLLMENT_DAYS_28,
+            ):
+                raise ValueError(
+                    "Covariate adjustment must be done using pre-treatment analysis period"
+                )
```

Comment:
Might be helpful to include the options in the error?
```suggestion
            preenrollment_periods = [
                parser_metric.AnalysisPeriod.PREENROLLMENT_WEEK,
                parser_metric.AnalysisPeriod.PREENROLLMENT_DAYS_28,
            ]
            if covariate_period not in preenrollment_periods:
                raise ValueError(
                    f"Covariate adjustment must be done using pre-treatment analysis period: (one of: {preenrollment_periods})"
                )
```

Reply:
Done